internal/api/automations.go

```diff
@@ -0,0 +1,169 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+// AutomationsClient is a client for working with automations.
+type AutomationsClient interface {
+	Get(ctx context.Context, id uuid.UUID) (*Automation, error)
+	Create(ctx context.Context, data AutomationUpsert) (*Automation, error)
+	Update(ctx context.Context, id uuid.UUID, data AutomationUpsert) error
+	Delete(ctx context.Context, id uuid.UUID) error
+}
+
+// Automation represents an automation response.
+type Automation struct {
+	BaseModel
+	AutomationUpsert
+	AccountID   uuid.UUID `json:"account_id"`
+	WorkspaceID uuid.UUID `json:"workspace_id"`
+}
+
+// AutomationUpsert is the data needed to create or update an automation.
+type AutomationUpsert struct {
+	Name             string   `json:"name"`
+	Description      string   `json:"description"`
+	Enabled          bool     `json:"enabled"`
+	Trigger          Trigger  `json:"trigger"`
+	Actions          []Action `json:"actions"`
+	ActionsOnTrigger []Action `json:"actions_on_trigger"`
+	ActionsOnResolve []Action `json:"actions_on_resolve"`
+}
+
+// Trigger defines the triggering conditions on an Automation.
+// On the API, a Trigger is a polymorphic type and can be represented
+// by several schemas based on the `type` attribute.
+// Here, we'll combine all possible attributes for each type
+// and make them optional.
+type Trigger struct {
+	Type string `json:"type"`
+
+	// For EventTrigger
+	Match        *ResourceSpecification `json:"match,omitempty"`
+	MatchRelated *ResourceSpecification `json:"match_related,omitempty"`
+	Posture      *string                `json:"posture,omitempty"`
+	After        []string               `json:"after,omitempty"`
+	Expect       []string               `json:"expect,omitempty"`
+	ForEach      []string               `json:"for_each,omitempty"`
+	Threshold    *int                   `json:"threshold,omitempty"`
+	Within       *string                `json:"within,omitempty"` // Duration string
```

Comment:
is this saying "Within" is similar to a duration? almost looks like this was going to be another struct field that lost its new line but wanted to check

Reply:
yeah the former - it's a string that reflects a duration